packages/monitor-v2/src/monitor-og/SnapshotVerification.ts

```diff
@@ -104,27 +111,59 @@ const getGraphqlData = async (ipfsHash: string, url: string): Promise<GraphqlDat
     }
   `);
   try {
-    return await request<GraphqlData, { ipfsHash: string }>(url, query, { ipfsHash });
-  } catch (error) {
-    return new Error(JSON.stringify(error));
+    return await retry(
+      async () => {
+        return await request<GraphqlData, { ipfsHash: string }>(url, query, { ipfsHash });
+      },
+      {
+        ...retryOptions,
+        onRetry(error) {
+          // We only log the error from the last retry if all attempts failed.
+          lastTryError = error;
+        },
+      }
+    );
+  } catch {
+    // We only try retry above. If that failed, there must be an error assigned to lastTryError.
+    assert(lastTryError instanceof Error);
+    return lastTryError;
   }
 };
 
 // We don't want to throw an error if the IPFS request fails for any reason, so we return a stringified Error object
 // instead that will be logged by the bot.
-const getIpfsData = async (ipfsHash: string, url: string): Promise<IpfsData | Error> => {
+const getIpfsData = async (ipfsHash: string, url: string, retryOptions: RetryOptions): Promise<IpfsData | Error> => {
+  let lastTryError: Error | undefined;
   try {
-    const response = await fetch(`${url}/${ipfsHash}`);
-    if (!response.ok) {
-      return new Error(`Request on ${response.url} failed with status ${response.status}`);
-    }
+    const response = await retry(
```

Comment:
nit: similar suggestion as above

Reply:
Yes, we can avoid storing last retry error. Though, I keep try block as we want to also try parsing the fetch response